src/sortable.js

```diff
@@ -23,116 +23,110 @@ angular.module('ui.sortable', [])
           var savedNodes;
 
           function combineCallbacks(first, second){
-            var firstIsFunc = typeof first === 'function';
-            var secondIsFunc = typeof second === 'function';
-            if(firstIsFunc && secondIsFunc) {
-              return function() {
-                first.apply(this, arguments);
-                second.apply(this, arguments);
-              };
-            } else if (secondIsFunc) {
+            if (typeof second === 'function') {
+              if (typeof first === 'function') {
+                return function() {
+                  first.apply(this, arguments);
+                  second.apply(this, arguments);
+                };
+              }
               return second;
             }
             return first;
           }
 
+          function wrappersHelper(inner) {
+            return function (e, item) {
+              var oldItemSortable = item.sortable;
+              var index = getItemIndex(item);
+              item.sortable = {
+                model: ngModel.$modelValue[index],
+                index: index,
+                source: item.parent(),
+                sourceModel: ngModel.$modelValue,
+                _restore: function () {
+                  delete item.sortable;
+                  item.sortable = oldItemSortable;
+                }
+              };
+            
+              var innerResult = inner.apply(this, arguments);
+              item.sortable._restore();
+              item.sortable._isCustomHelperUsed = item !== innerResult;
+              return innerResult;
+            };
+          }
+
           function getSortableWidgetInstance(element) {
             // this is a fix to support jquery-ui prior to v1.11.x
             // otherwise we should be using `element.sortable('instance')`
             var data = element.data('ui-sortable');
             if (data && typeof data === 'object' && data.widgetFullName === 'ui-sortable') {
               return data;
             }
-            return null;
           }
 
           function patchSortableOption(key, value) {
             if (callbacks[key]) {
               if( key === 'stop' ){
-                // call apply after stop
-                value = combineCallbacks(
-                  value, function() { scope.$apply(); });
-
-                value = combineCallbacks(value, afterStop);
+                value = combineCallbacks(value, function (e, ui) {
+                  // call apply after stop
+                  scope.$apply();
+                  ui.item.sortable._destroy();
+                });
               }
               // wrap the callback
               value = combineCallbacks(callbacks[key], value);
-            } else if (wrappers[key]) {
-              value = wrappers[key](value);
+            } else if ( key === 'helper' && typeof value === 'function') {
```

Comment:
I preferred the more generic approach, since this would require less effort in case we need to wrap any other callback.